### ISSUE TITLE
Lag Despite Removing Mod (Performance Optimizer)

### ISSUE THREAD

**Issue by Dragoonin** [AnthonyChenevier/EnhancedGrowthVatLearning#8]
Error Logs - 
https://gist.github.com/HugsLibRecordKeeper/945c9c79e7f8410743b20b8f23bc9cc3

Error Thrown Shown In Dev Mode - 

Exception ticking GrowthVat6114085 (at (144, 0, 185)): System.NullReferenceException: Object reference not set to an instance of an object.
[Ref BEA8AADC] Duplicate stacktrace, see ref for original
UnityEngine.StackTraceUtility:ExtractStackTrace ()
(wrapper dynamic-method) MonoMod.Utils.DynamicMethodDefinition:Verse.Log.Error_Patch3 (string)
Verse.TickList:Tick ()
(wrapper dynamic-method) MonoMod.Utils.DynamicMethodDefinition:Verse.TickManager.DoSingleTick_Patch3 (Verse.TickManager)
Verse.TickManager:TickManagerUpdate ()
(wrapper dynamic-method) MonoMod.Utils.DynamicMethodDefinition:Verse.Game.UpdatePlay_Patch2 (Verse.Game)
Verse.Root_Play:Update ()

Save File - 
https://www.dropbox.com/scl/fi/2oa83nbvhf5dogoqog3l5/Autosave-8.rws?rlkey=16am5f6jqn8nov7fdtbv5ogyk&st=wra5shzb&dl=0

Steps to Reproduce - 
1. Have Growth Vats
2. Let Hediff Overclocked Vat Learning Reach 100% (It does not repeat like normal vat learning.)

Apologize if I am reporting too much, just trying you know when issues occur.
I also removed performance optimizer, but still ran into this issue, it seems to occur when pawns are in the learning stage of growing.
1-3 as a baby there is no issues, and 13-18 there are no issues seemingly, but once growth stages begins it has issues.

**Comment by AnthonyChenevier** [comment]
Thanks for the report. I'll check out your save file and see if any of the mods you have are messing with mine - As I've developed this with a bunch of mods installed (including large popular mods) this *shouldn't* be a major issue, but then again a lot of mods have updated/been added in the time I've been away from the game, so having a save file to test with could be very useful. 

More reports = good, btw. Wish more peeps had done so in the beta!

**Comment by AnthonyChenevier** [comment]
@Dragoonin Ok so I've done some extensive testing and the point that it's breaking for me is the built-in RW "LearningUtility.LearningRateFactor(Pawn pawn)" function which I don't touch and isn't a problem in most cases. I also noticed that the Cleopatra pawn was erroring, but that may have been because I loaded your save without most of your mods...
There may be a way to salvage your save though - I was able to get it working again by simply loading the save, ejecting the pawns and reentering their vats (still in pause after initial loading). After that everything worked as expected - they even got the level ups they should have gotten.

EDIT: just to add, it looks like it may have been having difficulty loading the "learning rate" def, but that's only a guess as my debugging tools can't look at the native code run-state
![image](https://github.com/user-attachments/assets/cbf36d6e-13ab-4873-9e9f-9e4cf39b592e)
